projects/packages/videopress/src/client/lib/token-bridge/index.ts

```diff
@@ -24,11 +24,43 @@ const { videopressAjax } = window;
 type TokenBrigeEventProps = {
 	event: 'videopress_token_request';
 	guid: VideoGUID;
+	subscriptionPlanId?: number;
 	requestId: string;
 	origin: Origin;
 	isRetry?: boolean;
 };
 
+/**
+ * Quick docReady implementation.
+ *
+ * @param {CallableFunction} fn - Call this on ready.
+ */
+function ready( fn: CallableFunction ) {
+	if ( document.readyState !== 'loading' ) {
+		fn();
+	} else {
+		document.addEventListener( 'DOMContentLoaded', fn );
+	}
+}
+
+/**
+ * Check if the guid has an associated subscriptionPlanId.
+ *
+ * @param {VideoGUID} guid - The guid.
+ * @returns {Promise} promise.
+ */
+function getSubscriberPlanIdIfExists( guid: VideoGUID ): Promise< number > {
+	return new Promise( function ( resolve ) {
```

Comment:
I think it would make things way clearer if ready() was returning a premise instead of a function. You're mixing callbacks and promises and it never gets good. What about this approach:

```

       
async function getSubscriberPlanIdIfExists( guid: VideoGUID ) : Promise<number> {  
     // Can also be defined outside
     const ready = new Promise( ( resolve ) => document.readyState != "loading" ? 
       resolve() : 
       document.addEventListener("DOMContentLoaded", resolve ) );
    
       await ready();
       if ( ! window.__guidsToPlanIds ) {
	   return 0;
	}
	return window.__guidsToPlanIds[ guid ] || 0;
}

```
(syntax might be off as I wrote everything here directly)

Reply:
nice idea! Refactoring